module/zfs/zcp_get.c

```diff
@@ -497,8 +496,8 @@ get_zap_prop(lua_State *state, dsl_dataset_t *ds, zfs_prop_t zfs_prop)
 boolean_t
 prop_valid_for_ds(dsl_dataset_t *ds, zfs_prop_t zfs_prop)
 {
-	int error;
-	zfs_type_t zfs_type;
+	int error = 0;
```

Comment:
nit: It looks like you could declare `error` below in the one place it's used.

```c
	int error = get_objset_type(ds, &zfs_type);
```

Reply:
Good point!